### PR TITLE
feat: Add the picks stats endpoint

### LINKED ISSUE #14: Implement endpoint to get the number of likes for an item
Endpoint: `GET /picks/:itemId/stats` (OPTIONAL AUTH)
- If the request is not authenticated, it should return only the number of likes.
Expected response: `{ count: number }`
- If the request is authenticated, it should return the number of likes and a flag indicating if the item is already liked by the authenticated user.
Expected response: `{ likedByUser: boolean, count: number }`

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR adds the picks stats endpoint. This endpoint retrieves the amount of favorites an item has and if the user favorited or not the item.

Closes #14

### CODE REVIEW

File: test/unit/picks-component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  })
+
+  describe("and the user address parameter is set", () => {
+    beforeEach(async () => {
+      options.userAddress = "aUserAddress"
+      dbQueryMock.mockResolvedValueOnce({ rows: [{ counts: 1000, likedByUser: true }] })
+      result = await picksComponent.getPickStats(itemId, options)
+    })
+
+    it("should not check in the query if the user has liked the item", () => {
```
Comment (meelrossi) [inline test/unit/picks-component.spec.ts:83]:
```suggestion
    it("should check in the query if the user has liked the item", () => {
```

is this should check? or im not understanding correctly

File: test/unit/picks-component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      expect(dbQueryMock).toBeCalledWith(
+        expect.objectContaining({
+          text: expect.stringContaining("SELECT COUNT(*) counter FROM favorites.picks"),
+          values: expect.arrayContaining([options.userAddress, itemId]),
+        })
+      )
+      expect(dbQueryMock).toBeCalledWith(
+        expect.objectContaining({
+          text: expect.stringContaining("GROUP BY (hasLiked.counter)"),
+          values: expect.arrayContaining([options.userAddress, itemId]),
```
Comment (meelrossi) [inline test/unit/picks-component.spec.ts:98]: i think there is no need to add this here right? as it is tested in the previous expect

File: test/unit/picks-component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  })
+
+  describe("and the user address parameter is not set", () => {
+    beforeEach(async () => {
+      options.userAddress = undefined
+      dbQueryMock.mockResolvedValueOnce({ rows: [{ counts: 1000 }] })
+      result = await picksComponent.getPickStats(itemId, options)
+    })
+
+    it("should check in the query if the user has liked the item", () => {
```
Comment (meelrossi) [inline test/unit/picks-component.spec.ts:115]: and here is not i believe
```suggestion
    it("should not check in the query if the user has liked the item", () => {
```

File: test/unit/picks-component.spec.ts (hunk 2 above)
Comment (LautaroPetaccio) [reply test/unit/picks-component.spec.ts:98]: Removed it.

File: test/unit/picks-component.spec.ts (hunk 1 above)
Comment (LautaroPetaccio) [reply test/unit/picks-component.spec.ts:83]: You're right! Changed it.

File: test/unit/picks-component.spec.ts (hunk 3 above)
Comment (LautaroPetaccio) [reply test/unit/picks-component.spec.ts:115]: Changed it as well.

**Review by meelrossi — APPROVED** [review]

**Pull request merged**